### PR TITLE
Add Buybacker to graph entities

### PR DESCRIPTION
**Pull request by abtheo**
Hey @d0c0q , the docker image for the subgraph works great! Sorry it took so long to get back to you, but everything is going smoothly 🙂

This PR adds Buybacks as an event to the subgraph. However, it currently **does not add buybacks to the protocolMetrics**. 

I believe we should call the `updateProtocolMetrics` function whenever a Buyback event is caught by the `handleBuyback(event: BuybackEvent)` handler. Unfortunately, I couldn't figure out a clean way to do this without passing the entity / event itself into the protocol metrics 😅

My other thought was that perhaps we only need the Buybacks as an entity? Then we could construct a GraphQL query on the front-end to get all the events and order them by time, e.g.

`
{
    buybacks(orderBy: timestamp, orderDirection: desc) {
        id,
        timestamp,
        token,
        tokenAmount,
        clamAmount,
    }
}
`

But I'm not sure what the best way to go is. Any ideas?

### CODE REVIEW

**Comment by d0c0q** [comment]
`updateProtocolMetrics` will truncate timestamp to day precision and collect metrics which at this moment. It will trigger when transferring or depositing event happened.

If you just want to make a graph for buyback event, I think your idea is good enough.

File: src/OtterBuybacker.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  let transaction = loadOrCreateTransaction(event.transaction, event.block)
+  let entity = new Buyback(transaction.id)
+  entity.token = event.params.token
+  entity.tokenAmount = event.params.tokenAmount
+  entity.clamAccount = event.params.clamAmount
+  entity.timestamp = transaction.timestamp
+  entity.transaction = transaction.id
+  entity.save()
+
+  //TODO: updateProtocolMetrics with Buyback amount ??
```
Comment (d0c0q) [inline src/OtterBuybacker.ts:16]:
```suggestion
```

**Comment by otterclamking** [comment]
conflict @abtheo

**Comment by otterclamking** [comment]
@abtheo the `protocolMetrics` is the accumulated result, so if we need to display the sum of the buyback, we need to write it to `protocolMetrics` . The buyback entity only presents the value of buyback everytime.

**Pull request closed without merging**